dd-sdk-android-ndk/src/main/cpp/utils/backtrace-handler.cpp

```diff
@@ -94,25 +91,28 @@ namespace {
 
 }
 
-namespace backtrace {
-
-    std::string generate_backtrace() {
-        // define the buffer which will hold pointers to stack memory addresses
-        uintptr_t buffer[STACK_SIZE];
-        // we will now unwind the stack and capture all the memory addresses up to STACK_SIZE in
-        // the buffer
-        const size_t captured_stacksize = capture_backtrace(buffer, STACK_SIZE);
-        std::string backtrace;
-        for (size_t idx = 0; idx < captured_stacksize; ++idx) {
-            // we will iterate through all the stack addresses and translate each address in
-            // readable informationdsadsa
-            get_info_from_address(buffer[idx], &backtrace);
+bool copyString(const std::string &str, char *ptr, size_t max_size) {
+    if (str.size() >= max_size) return false;
```

Comment:
this version of the function will juste refuse to copy anything if the max_size is not large enough.
Alternatively you could do:
```
size_t copy_size = std::min(str.size(), max_size - 1);
memcpy(ptr, str.data(), copy_size);
ptr[copy_size] = '\0';
return str.size() < max_size;
```

This would truncate rather than fail.